### DISCUSSION TITLE
How does yabai decide where to open the next window?

### DISCUSSION THREAD

**Discussion by yannxaver** [asmvik/yabai#2473] [Q&A]
First time using a tiling window manager.

The only consistency I see so far: 
- If I have one window open and open a second one, it will open as a column next to the first one. 

After that, it seems to get arbitrary. The third window might open in a third column or as a second row in the first column or as a second row in the second column. 

It's not clear to me how this gets decided. At first, I thought the column for the third window gets picked based on which column is selected when opening it, but this also doesn't seem to be the case.

**Comment by Bellavene** [comment]
It is clear that a new window will open by halving the active window. In fact, there is no setting to change that.

**Comment by pabroux** [comment]
Hello @yannxaver,

Yes, it seems to be arbitrary after the third windows when you don't instruct yabai what to do (see below). After using yabai for at least 5 years, it has never bothered me.  Why do you want to know that?

By the way, if you want to tell yabai where the future new window should be, use one of the followings:
```
yabai -m window --insert west
yabai -m window --insert south
yabai -m window --insert north
yabai -m window --insert east
```
These commands will create an orange area over your active window corresponding to the position on which you want to insert a new window. Then, when you press `CMD-N` to create a new window, the new window will take the position of the orange area.

_p.s: If you need inspiration, feel free to check my [skhdrc](https://github.com/pabroux/unvX/blob/master/module/skhd/skhdrc) file in my dotfile manager._
_p.s. bis: If this message answers your question, please make it as “answered"._

**Comment by Bellavene** [comment]
Also check the new setting `window_insertion_point`
And there is `split_type`
It is possible to script the opening of all new windows in one separate column only. If you want more control, you will need bash/zsh scripting anyway. And to fix some window detection bugs that are causing inconsistencies with some apps.
